packages/hierarchies-react/api/presentation-hierarchies-react.api.md

```diff
@@ -206,7 +206,7 @@ export const TreeNodeRenderer: ForwardRefExoticComponent<TreeNodeRendererProps &
 // @alpha (undocumented)
 interface TreeNodeRendererOwnProps {
     actions?: Array<(node: PresentationHierarchyNode) => TreeItemAction>;
-    getIcon?: (node: PresentationHierarchyNode) => string | ReactElement | undefined;
+    getDecorations?: Array<(node: PresentationHierarchyNode) => ReactElement | undefined>;
```

Comment:
Is there any benefits taking array of callback producing react elements instead of single callback that produces array of elements?

Reply:
I guess there is no big difference, only one being is that if we decide to provide decorations it would be more readable implementation on consumer end. Not sure which one would be better

Reply:
> if we decide to provide decorations it would be more readable implementation on consumer end

Not sure I understand what you mean here. I personally like the single callback approach better.

Reply:
Basically having one callbacks will in the end produce the same structure as multiple callbacks, e.g. you have (node) => [functionForColorPicker(node), functionForIcon(node)] ==> [functionForColorPicker, functionForIcon]. In the case for TreeWidget we might need to expose out icons function if the user wants to have icons and colorpicker position fluid not static in right for the icons at all times.

Reply:
I think we should make `getDecorations` API match `TreeItem.unstable_decorations`. E.g. `getDecorations?: (node) => ReactNode`. 

Also instead of functions for color picker and icon we could provide components:
```tsx
getDecorations: (node) => <>
  <ColorPicker node={node} {...otherProps} />
  <SomeTreeIcon node={node} />
</>
```

Reply:
Yeah forgot that they would be components, changed to a single callback and react node so the consumer can decide if to use an array or fragments.